sample_project/Assets/SampleViewer/Samples/Geometry/Geometries.cs

```diff
@@ -0,0 +1,501 @@
+		featurePoints.Add(bottomRightMarker);
+		Interpolate(bottomRightMarker, bottomLeftMarker, featurePoints);
+		featurePoints.Add(bottomLeftMarker);
+		Interpolate(bottomLeftMarker, topLeftMarker, featurePoints);
+		featurePoints.Add(topLeftMarker);
+
+		RenderLine(ref featurePoints);
+		RebaseLine();
+	}
+
+	private void SetSurfacePlacement(GameObject marker, double offset)
+	{
+		var locationComponent = marker.GetComponent<ArcGISLocationComponent>();
+		locationComponent.SurfacePlacementMode = ArcGISSurfacePlacementMode.RelativeToGround;
+		locationComponent.SurfacePlacementOffset = offset;
+	}
+
+	private void CreateandCalculatePolygon()
+	{
+		var spatialReference = new ArcGISSpatialReference(3857);
+		var polygonBuilder = new ArcGISPolygonBuilder(spatialReference);
+
+		foreach (var stop in stops)
+		{
+			var location = stop.GetComponent<ArcGISLocationComponent>().Position;
+			polygonBuilder.AddPoint(location);
+		}
+
+		var polygon = polygonBuilder.ToGeometry();
+
+		geodeticAreaPolygon = ArcGISGeometryEngine.AreaGeodetic(polygon, currentAreaUnit, ArcGISGeodeticCurveType.Geodesic);
+		UpdateDisplay();
+	}
+	private void Interpolate(GameObject start, GameObject end, List<GameObject> featurePoints)
+	{
+
+		float lengthOfLine = Vector3.Distance(start.transform.position, end.transform.position);
+		float n = Mathf.Floor(lengthOfLine / InterpolationInterval);
+		double dx = (end.transform.position.x - start.transform.position.x) / n;
+		double dy = (end.transform.position.y - start.transform.position.y) / n;
+		double dz = (end.transform.position.z - start.transform.position.z) / n;
+
+		var previousInterpolation = start.transform.position;
+
+		// Calculate n-1 intepolation points/n-1 segments because the last segment is already created by the end point.
+		for (int i = 0; i < n - 1; i++)
+		{
+			GameObject nextInterpolation = Instantiate(InterpolationMarker, arcGISMapComponent.transform);
+
+			// Calculate transform of nextInterpolation point.
+			float nextInterpolationX = previousInterpolation.x + (float)dx;
+			float nextInterpolationY = previousInterpolation.y + (float)dy;
+			float nextInterpolationZ = previousInterpolation.z + (float)dz;
+			nextInterpolation.transform.position = new Vector3(nextInterpolationX, nextInterpolationY, nextInterpolationZ);
+
+			// Set default location component of nextInterpolation point.
+			nextInterpolation.GetComponent<ArcGISLocationComponent>().enabled = true;
+			nextInterpolation.GetComponent<ArcGISLocationComponent>().Rotation = new ArcGISRotation(0, 90, 0);
+
+			var location = nextInterpolation.GetComponent<ArcGISLocationComponent>();
+			location.Position = arcGISMapComponent.EngineToGeographic(nextInterpolation.transform.position);
+
+			featurePoints.Add(nextInterpolation);
+			previousInterpolation = nextInterpolation.transform.position;
+		}
+	}
+
+	// Set height for point transform and location component.
+	private void SetElevation(GameObject stop)
```

Comment:
Can we avoid this by using surface placement?

Reply:
I was thinking exactly the same but I tried to remove `SetElevation` and it doesn't work for some reasons. In the image below I commented out the bottom left marker

![image](https://github.com/user-attachments/assets/d9e32a4b-7c4f-47f4-b852-e0db9baa2390)

Reply:
Another question is, do we need to use elevation at all in this sample? Followup question, does the geometry engine take into the difference in elevation since the shape is not a rectangle if the points are not on the same plane?

Reply:
Good question. I think the geometry engine draw envelops based on top-down view so it can always draws even the four vertices are not level. 

And the reason for using elevation is to place markers on terrain, otherwise they're invisible.

Reply:
@airaolagoitia I used `locationComponent.SurfacePlacementMode = ArcGISSurfacePlacementMode.RelativeToGround;` but it doesn't place markers on ground (I have a small offset to account for marker height), and still need to use raycasting to place it correctly. Any insights?

Reply:
I guess what I am wondering about is whether the area and distance measurements  take the elevation change into account. If that's not the case, we could remove the elevation source. That way, we wouldn't need any raycasts and wouldn't need to split the lines into smaller segments.

Reply:
Yet another question, is the geodesic area/distance correct if we are using a local map?

Reply:
That's a good point. https://developers.arcgis.com/javascript/latest/tutorials/find-length-and-area/ this says that if spatial ref is 3857, we should use geodesic area/distance. Both the measure and geometry scene has spatial ref of 3857 from viewstate logging

Reply:
From this [API ref](https://developers.arcgis.com/unity/api-reference/gameengine/geometry/arcgisgeometryengine/#distancegeodetic), `distancegeodetic` takes in two `ArcGISPoints` which has Z value, so the distance calculation accounts for z value, but it doesn't account for elevation since it has no knowledge of elevation. 

Therefore this sample takes elevation into account by letting user keep adding points and sum up all the small intervals. In the image below, when users add more points along the terrain, it adds up those intervals together which accounts for terrain (although the accuracy depends on how often users add points).

![image](https://github.com/user-attachments/assets/560ccdce-f34a-4099-bc6b-581f209191b3)